internal/operator-controller/controllers/clusterextensionrevision_controller.go

```diff
@@ -428,14 +441,53 @@ func (c *ClusterExtensionRevisionReconciler) removeFinalizer(ctx context.Context
 	return nil
 }
 
-func toBoxcutterRevision(rev *ocv1.ClusterExtensionRevision) (*boxcutter.Revision, []boxcutter.RevisionReconcileOption, []client.Object) {
-	previous := make([]client.Object, 0, len(rev.Spec.Previous))
-	for _, specPrevious := range rev.Spec.Previous {
-		prev := &unstructured.Unstructured{}
-		prev.SetName(specPrevious.Name)
-		prev.SetUID(specPrevious.UID)
-		prev.SetGroupVersionKind(ocv1.GroupVersion.WithKind(ocv1.ClusterExtensionRevisionKind))
-		previous = append(previous, prev)
+// ListPreviousRevisions returns active revisions belonging to the same ClusterExtension with lower revision numbers.
+//
+// Scenario: Given a ClusterExtensionRevision, find all related active previous revisions
+//   - When the revision has no owner label, return empty list (revision not properly labeled)
+//   - When looking up by owner label, filter out the current revision itself
+//   - When a revision is archived or being deleted, exclude it from results
+//   - When a revision has a higher or equal revision number, exclude it (not a previous revision)
+//   - Otherwise include active and paused revisions with matching owner
+func (c *ClusterExtensionRevisionReconciler) ListPreviousRevisions(ctx context.Context, rev *ocv1.ClusterExtensionRevision) ([]client.Object, error) {
+	ownerLabel, ok := rev.Labels[ClusterExtensionRevisionOwnerLabel]
+	if !ok {
+		// No owner label means this revision isn't properly labeled - return empty list
+		return nil, nil
+	}
+
+	revList := &ocv1.ClusterExtensionRevisionList{}
+	if err := c.Client.List(ctx, revList, client.MatchingLabels{
+		ClusterExtensionRevisionOwnerLabel: ownerLabel,
+	}); err != nil {
+		return nil, fmt.Errorf("listing revisions: %w", err)
+	}
+
+	previous := make([]client.Object, 0, len(revList.Items))
+	for i := range revList.Items {
+		r := &revList.Items[i]
+		if r.Name == rev.Name {
+			continue
+		}
+		// Skip archived or deleting revisions
+		if r.Spec.LifecycleState == ocv1.ClusterExtensionRevisionLifecycleStateArchived ||
+			!r.DeletionTimestamp.IsZero() {
+			continue
+		}
+		// Only archive revisions with lower revision numbers (actual previous revisions)
+		if r.Spec.Revision >= rev.Spec.Revision {
+			continue
```

Comment:
We still populate `Spec.Previous` even though we now use labels for discovery because:

- 1. **API Compatibility** - It's a published field in the CRD and marked immutable. Removing it would be a breaking change.
- 2. **Garbage Collection** - `setPreviousRevisions()` uses it to enforce the 5 revision limit and delete older archived revisions.
- 3. **Audit Trail** - Provides a historical record of the revision chain, useful for debugging and understanding upgrade history.

Reply:
I'm not sure about this one. I think we can violate `i` since we're still in experimental mode and breaking changes are allowed, we should probably update `ii` to use the List function anyway, and lastly, revisions are numbered, so the chain should be self-evident, no?

Reply:
I think you are 100% right and we can keep things simple here by removing it.